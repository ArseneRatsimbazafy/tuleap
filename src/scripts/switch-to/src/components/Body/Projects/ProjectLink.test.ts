/*
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import { Project } from "../../../type";
import ProjectLink from "./ProjectLink.vue";
import { createStoreMock } from "../../../../../vue-components/store-wrapper-jest";
import { State } from "../../../store/type";

describe("ProjectLink", () => {
    it("Displays the link to a project", () => {
        const wrapper = shallowMount(ProjectLink, {
            propsData: {
                project: {
                    is_public: true,
                    project_name: "Guinea Pig",
                    project_uri: "/pojects/gpig",
                } as Project,
            },
            mocks: {
                $store: createStoreMock({
                    state: {
                        are_restricted_users_allowed: true,
                    } as State,
                }),
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
