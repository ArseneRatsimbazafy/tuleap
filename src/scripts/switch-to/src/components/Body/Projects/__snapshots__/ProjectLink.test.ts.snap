// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLink Displays the link to a project 1`] = `
<div
  class="switch-to-projects-project"
>
  <a
    class="switch-to-projects-project-link"
    data-test="project-link"
    href="/pojects/gpig"
  >
    <i
      class="fa fa-fw switch-to-projects-project-icon fa-unlock"
    />
     
    <span
      class="switch-to-projects-project-label"
    >
      Guinea Pig
    </span>
  </a>
   
  <!---->
</div>
`;
